pkg/deploy/ec2/endpoint_service_manager.go

```diff
@@ -0,0 +1,265 @@
+package ec2
+
+import (
+	"context"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	ec2sdk "github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/algorithm"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/aws/services"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/deploy/tracking"
+	ec2model "sigs.k8s.io/aws-load-balancer-controller/pkg/model/ec2"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/networking"
+)
+
+// abstraction around endpoint service operations for EC2.
+type EndpointServiceManager interface {
+	Create(ctx context.Context, resES *ec2model.VPCEndpointService) (ec2model.VPCEndpointServiceStatus, error)
+
+	Update(ctx context.Context, resES *ec2model.VPCEndpointService, sdkES networking.VPCEndpointServiceInfo) (ec2model.VPCEndpointServiceStatus, error)
+
+	Delete(ctx context.Context, sdkES networking.VPCEndpointServiceInfo) error
+
+	ReconcilePermissions(ctx context.Context, permissions *ec2model.VPCEndpointServicePermissions) error
+}
+
+// NewdefaultEndpointServiceManager constructs new defaultEndpointServiceManager.
+func NewDefaultEndpointServiceManager(ec2Client services.EC2, vpcID string, logger logr.Logger, trackingProvider tracking.Provider, taggingManager TaggingManager, externalManagedTags []string) *defaultEndpointServiceManager {
+	return &defaultEndpointServiceManager{
+		ec2Client:           ec2Client,
+		vpcID:               vpcID,
+		logger:              logger,
+		taggingManager:      taggingManager,
+		trackingProvider:    trackingProvider,
+		externalManagedTags: externalManagedTags,
+	}
+}
+
+var _ EndpointServiceManager = &defaultEndpointServiceManager{}
+
+// default implementation for EndpointServiceManager.
+type defaultEndpointServiceManager struct {
+	ec2Client           services.EC2
+	vpcID               string
+	logger              logr.Logger
+	taggingManager      TaggingManager
+	trackingProvider    tracking.Provider
+	externalManagedTags []string
+}
+
+func (m *defaultEndpointServiceManager) Create(ctx context.Context, resSG *ec2model.VPCEndpointService) (ec2model.VPCEndpointServiceStatus, error) {
```

Comment:
Seems there's still a few references to `SG` rather than `ES` throughout this function.
```suggestion
func (m *defaultEndpointServiceManager) Create(ctx context.Context, resES *ec2model.VPCEndpointService) (ec2model.VPCEndpointServiceStatus, error) {
```